### ISSUE TITLE
b.cesm3_cam041_mom.B1850WcMOM.ne30_L58_t061.005

### ISSUE THREAD

**Issue by cecilehannay** [NCAR/amwg_dev#115]
**Description:**

Coupled with CAM-SE-L58 + MOM + CICE6 with BWsc1850 compset

Includes bugfix for the sea-ice to fix excessive fresh water (See https://github.com/NCAR/amwg_dev/discussions/109)

- compset: ``1850_CAM60%WCSC_CLM50%BGC-CROP_CICE_MOM6_MOSART_CISM2%GRIS-NOEVOLVE_SWAV_SESP_BGC%BDRD``
- res: ``ne30pg3_t061``
- SourceMods: available [here](https://github.com/NCAR/amwg_dev/tree/b.cesm3_cam041_mom.B1850WcMOM.ne30_L58_t061.005) 
- user_nl_*: available [here](https://github.com/NCAR/amwg_dev/tree/b.cesm3_cam041_mom.B1850WcMOM.ne30_L58_t061.005) 

**Case directory:**
Locally (if still available):
/glade/p/cesmdata/cseg/runs/cesm2_0/b.cesm3_cam041_mom.B1850WcMOM.ne30_L58_t061.005

On github:
https://github.com/NCAR/amwg_dev/tree/b.cesm3_cam041_mom.B1850WcMOM.ne30_L58_t061.005

**Sandbox:**
Locally (if still available):
/glade/work/hannay/cesm_tags/cesm3_cam6_3_041_MOM3

On github:
https://github.com/NCAR/amwg_dev/tree/cesm3_cam6_3_041_MOM3
hash: 5a56ad3

**Diagnostics:**
AMWG diags (if available)
[https://webext.cgd.ucar.edu/BWsc1850MOM/b.cesm3_cam041_mom.B1850WcMOM.ne30_L58_t061.005/atm/](https://webext.cgd.ucar.edu/BWsc1850MOM/b.cesm3_cam041_mom.B1850WcMOM.ne30_L58_t061.005/atm/)

**Contacts:**
@cecilehannay and @gustavo-marques

**Comment by cecilehannay** [comment]
After consulting with @tilmes and Mike, we use the ``1850_CAM60%WCSC_CLM50%BGC-CROP_CICE_MOM6_MOSART_CISM2%GRIS-NOEVOLVE_SWAV_SESP_BGC%BDRD`` for the current run. 

Note that there is no need to H2O external forcings to the WACCM-SC run. WACCM-SC has CH4 in the chemistry mechanism, so it calculates H2O production from CH4 oxidation. Therefore, we don’t want to add this as an external forcing.

**Comment by cecilehannay** [comment]
The run crashed in year 21 with the error:

```4389:
4389:FATAL from PE   789: mpp_sync_self: size_recv does not match of data received
4389:
4389:Image              PC                Routine            Line        Source             
4389:cesm.exe           00000000042F5606  Unknown               Unknown  Unknown
4389:cesm.exe           0000000003DC632D  mpp_mod_mp_mpp_er          68  mpp_util_mpi.inc
4389:cesm.exe           0000000003DA1C8D  mpp_mod_mp_mpp_sy         224  mpp_util_mpi.inc
4389:cesm.exe           0000000003F22640  mpp_domains_mod_m         513  mpp_group_update.h
4389:cesm.exe           0000000002A608D8  mom_domain_infra_        1152  MOM_domain_infra.F90
4389:cesm.exe           000000000312A066  mom_barotropic_mp        1772  MOM_barotropic.F90
4389:cesm.exe           0000000002A847C1  mom_dynamics_spli         808  MOM_dynamics_split_RK2.F90
4389:cesm.exe           000000000285FD0F  mom_mp_step_mom_d        1114  MOM.F90
4389:cesm.exe           0000000002857759  mom_mp_step_mom_          830  MOM.F90
4389:cesm.exe           000000000282B4C0  mom_ocean_model_n         627  mom_ocean_model_nuopc.F90
4389:cesm.exe           00000000027F4616  mom_cap_mod_mp_mo        1684  mom_cap.F90
```

I am not sure whether 
``mpp_sync_self: size_recv does not match of data received`` 
was problem with cheyenne or mom. so, I restarted the run. 

@gustavo-marques: could you have a look too?

**Comment by gustavo-marques** [comment]
Not sure if this is related to the crash, but I see an issue with salt conservation starting at 20/01/21 00:00:00.

**Comment by gustavo-marques** [comment]
Actually, the issue with salt conservation starts in year 11.

**Comment by cecilehannay** [comment]
Should I continue the run? Or is it a show stopper?

**Comment by gustavo-marques** [comment]
Let's continue the run.

**Comment by gustavo-marques** [comment]
@dabail10: can you please check the ice.log to see if there are any conservation issues in CICE6?

**Comment by dabail10** [comment]
The salt and water are fine, but the shortwave looks off.

arwt incoming sw (W)   =    6.85618158455401250E+13   1.43538638248901375E+14
arwt absorbed sw (W)   =    6.94156657060417812E+13   1.43187650989964625E+14
arwt swdn error        =    1.23005355032898781E-02  -2.45123972989368718E-03

These errors should be closer to zero. Is there something going on with nextsw_cday and when the radiation is computed in the atmosphere?

**Comment by cecilehannay** [comment]
@JulioTBacmeister: could there be a problem with the atmosphere?

**Comment by dabail10** [comment]
I have not run the G case yet. I will try to set that up tomorrow. This was fine in a B case I did with CICE6 in cesm2.1.

**Comment by adamrher** [comment]
> These errors should be closer to zero. Is there something going on with nextsw_cday and when the radiation is computed in the atmosphere?

I hope there's not a problem w/ nextsw_cday. I spent a lot of time making sure it's correct using the new physics ordering. By correct, I mean that it reflects the zenith angle of the radiative fields actually being sent to the coupler. However, in the new physics ordering here, the radiation is less accurate than the old ordering. We no longer compute radiation just before sending the radiative fluxes to the coupler. They are now computed at the end of the physics time-loop in the prior time-step. Although in both cases its only computed every other time-step. So I might expect an increase in errors but I wouldn't expect it will cause near zero sw errors in the prior ordering, to goto E-2 errors like @dabail10 is showing, using the new physics ordering. but it could be something else I did wrong that I hadn't thought about.

**Comment by adamrher** [comment]
yes, the radiation time-steps are still 1 hour. nextsw_cday is used to sync up the albedo's in CLM, so hopefully if that's correct, everything falls in line.

**Comment by cecilehannay** [comment]
John Fasullo would like to add a few variables for future runs. 
- He would like us to add monthly TMQ, SOLIN, FLDS in the h0 file.  
- In the ocean, he is missing TEMP/MOC/SSH/HMXL but perhaps there are new variable names for MOM6? (@gustavo-marques?)

**Comment by gustavo-marques** [comment]
When possible, we use the Climate Model Output Rewriter (CMOR) convention for the MOM variables. 

The netCDF files listed below are located at /glade/scratch/hannay/archive/b.cesm3_cam041_mom.B1850WcMOM.ne30_L58_t061.005/ocn/hist/

* variable TEMP from POP (3D potential temperature) is variable `thetao` in MOM and can be found in the following files:
b.cesm3_cam041_mom.B1850WcMOM.ne30_L58_t061.005.mom6.h_\*.nc **(monthly averages regridded to the World Ocean Atlas 2009 geopotential vertical grid)**
b.cesm3_cam041_mom.B1850WcMOM.ne30_L58_t061.005.mom6.hm_\*.nc **(monthly averages in the native z\* vertical grid)**

* MOM does not output MOC. Rather, we output the meridional mass transport (kg s-1) due to advection (variable `vmo`), GM parameterization (variable `vhGM`), and the Restratify Mixed Layer parameterization (variable `vhml`). One needs to compute the MOC streamfunction offline by 1) adding these variables, 2) summing over the zonal direction, and 3) integrating vertically. I am happy to provide Python scripts for doing this if needed. These variables can also be found in the files listed above (*.h and *.hm, which are in z-space) and also in files b.cesm3_cam041_mom.B1850WcMOM.ne30_L58_t061.005.mom6.rho2_\*.nc (which are monthly averages in sigma2-space).

* variable HMXL in POP is variable `mlotst` in MOM; `SSH` has not changed. Both variables can be found in files:
b.cesm3_cam041_mom.B1850WcMOM.ne30_L58_t061.005.mom6.hm_\*.nc **(monthly averages)**
b.cesm3_cam041_mom.B1850WcMOM.ne30_L58_t061.005.mom6.sfc_\*.nc **(daily averages)**
